src/third-party/gutenberg.stories.mdx

```diff
@@ -0,0 +1,568 @@
+import { Story, Preview, Meta } from '@storybook/addon-docs/blocks';
+
+<Meta title="Third-Party/Gutenberg" />
+
+# WordPress Gutenberg Blocks
+
+The Wordpress Gutenberg editor is a block-based page builder. The editor comes with
+a variety of core blocks. Of those, a subset automatically generate markup with one
+or more `wp-block-`-prefixed classes. These are the blocks documented below.
+
+Each block has controls that can modify the generated output. Where relevant, the
+additional markup and/or classes that may be generated are also documented.
+
+### Editor Controls
+
+Block editor controls vary, but many have Styles and Color settings. The effects
+of other controls are described in specific block stories.
+
+#### Styles
+
+These controls offer 2 or more options for particular stylistic variations, and
+the ability to set one as a default for new blocks of the same type. For example,
+images can have a circular mask or not. These options add an `is-style-`-type class
+to the block.
+
+#### Color settings
+
+These controls enable setting of the block's `background-color` and `color`.
+
+The base class for a block with a main (background) color is `has-background` and
+selecting a text color adds a `has-text-color` class. If one of a set of predefined
+colors is chosen, additional classes are added, such as `has-subtle-background-background-color`
+for `background-color` or `has-secondary-color` for text.
+
+If user-defined colors are chosen, instead, inline styles will set `color` and/or
+`background-color`.
+
+## Audio
+
+The audio block wraps an `audio` element and an optional `figcaption` element in
+a `figure` with class `wp-block-audio`
+
+<Preview>
+  <Story name="core/audio" height="150px">{`
+    <figure class="wp-block-audio type">
+      <audio controls="" src=""></audio>
+      <figcaption>My audio clip</figcaption>
+    </figure>
+  `}</Story>
+</Preview>
+
+## Button
+
+The button block creates one or more `div`s with class `wp-block-button`, each containing
+an `a` element with a class of `wp-block-button__link`.
+
+An additional container holds each button block (one or more buttons), and has
+a class of `wp-block-buttons`.
+
+If chosen, a button can have only an outline (no fill color), which adds the `is-style-outline`
+class on the `.wp-block-button` element.
+
+Controls allow each button to have custom text color, background color or gradient,
+and border radius.
+
+<Preview>
+  <Story name="core/button" height="100px">
+    {`<div class="wp-block-buttons">
+        <div class="wp-block-button">
+          <a class="wp-block-button__link" href="https:/end wp-themes.com/twentytwenty/" style="border-radius:50px" target="_blank" rel="noopener noreferrer">
+            Button
+          </a>
+        </div>
+    </div>`}
+  </Story>
+</Preview>
+
+## Code
+
+The code block adds the class `wp-block-code` to a `pre` tag wrapping content inside
+`code` tags.
```

Comment:
I'm not sure it's necessary to describe in prose the markup structure and attribute contents for each pattern?

Reply:
I wonder if we should discuss my overall approach to story descriptions here. I didn't think mine was necessarily the best, but felt like some basic introduction was a good idea for all of them, and that for the simplest patterns, this approach was just a way to be consistent across descriptions. I realize that for something like the `code` block, probably no description is needed at all.

In this case, would you even include a description, and if so, would it just be something like `Provides a block-level class name for styling code snippets`?